### ISSUE TITLE
Rule proposal: Enforce numeric separators style

### ISSUE THREAD

**Issue by noftaly** [sindresorhus/eslint-plugin-unicorn#828]
Numeric separators are a new JavaScript feature that has [reached stage 4](https://github.com/tc39/proposal-numeric-separator) and allow numbers to be written like this:
```js
1_000_000_000
101_475_938.38 
0.000_001
1e10_000

// It also works with binary, hexadecimal, BigInts, Octals...
0b1010_0001_1000_0101
0xA0_B0_C0
9_223_372_036_854_775_807n
0o1234_5670
```

It would be great to have a rule to enforce its usage.
We could, for example, enforce grouping hexadecimal by 2, binary and octal by 4, numbers and BigInts by 3... I think that this preferred group length should be configurable.

It would also be great to have some more options such as:
- Only apply this rule if there is more than a certain number of digits. Example, if we disable the rule for less than 5 digits, then `1_024` will fail and `1024` will pass, but if we put it to 0, then `1_024` will pass and `1024` will fail
- The "preferred group length" could also be used to determine if a group is not of a valid length. It will check all non-leading groups, so in the case of `1_2_34_567`, it will be `2`, `34`, `567`, and see if it matches the preferred group length, here 3. If we fix this, we get `1_234_567`, which satisfy the "preferred group length" rule

With those two things in mind, maybe a configuration like this would be appropriate:
```js
// General:
{
    type: [minThreshold, preferedGroupLength],
}

// Example:
{
  hexadecimal: [0, 2],
  binary: [0, 4],
  octal: [0, 4],
  bigint: [5, 3],
  number: [5, 3],
}
```

~~I'm willing to try to make a PR for that, although this would be my first time creating an ESLint rule...~~

EDIT: Sorry for all the edits but I keep having new/better ideas, and I struggle to formulate them 😅

EDIT 2: I've created the rule and it works pretty well. [Here is the doc](https://gist.github.com/noftaly/23207a7c4dd04fdb0bedcdda3cfb3b0f) if you want to see the final result. I'll make a PR only if you guys are in favor of this. There are ~60 tests, with all scenarios I could imagine. The integration test was successful too.

**Comment by fregante** [comment]
I definitely agree that there should be a rule to refuse `10_0000` (unless you're korean) and the threshold idea also makes sense because below 1000000, separators aren't necessarily super useful.

**Comment by sindresorhus** [comment]
This sounds good to me. I think the `[0, 2]` should be an options-object instead (`{minimumThreshold: 0, preferedGroupLength: 2})`.

**Comment by sindresorhus** [comment]
Accepted. PR welcome.

**Comment by noftaly** [comment]
Thanks! Glad it has been accepted 😄
I just have one question before I make a PR:
What to do with decimals? `3.000_000_1` or `3.0_000_001`? `3.100_000_0` or `3.1_000_000`? Should both be accepted or only one? Then should it be configurable?
